src/item/enchantment/ItemEnchantmentTagRegistry.php

```diff
@@ -93,9 +93,9 @@ public function unregister(string $tag) : void{
 
 		unset($this->tagMap[$tag]);
 
-		foreach($this->tagMap as $tmpTag => $nestedTags){
+		foreach($this->tagMap as &$nestedTags){
```

Comment:
Foreach by-reference is generally avoided due to the ease of accidentally corrupting whatever is in the main array (e.g. if the same variable name is used in a later loop)

Reply:
What else is there to do? phpstan gives an error when I use a key in foreach (because the key is of string type, and it supposedly can be implicitly converted to int).

Reply:
Maybe do unset($nestedTags) after foreach?

Reply:
`foreach(Utils::stringifyKeys($array) as $key => $value){ ... }`

It should tell you that in a `Tip:` below the error.

Reply:
> because the key is of string type, and it supposedly can be implicitly converted to int

https://3v4l.org/TDOWjP